### PR TITLE
add is*Loaded methods

### LINKED ISSUE #509: Different gamma value?
Per chat with @ansis, experiment with different [gamma](http://mapbox-kkaefer.s3.amazonaws.com/sdf/index.html) values ([currently hardcoded](https://github.com/mapbox/mapbox-gl-js/blob/65b084c36ad3d7fda08c3d60314a1a8edfdc95d4/js/render/drawtext.js#L93)) in the [`symbol-render-type`](https://github.com/mapbox/mapbox-gl-js/tree/symbol-render-type) branch to possibly address [SDF quality issues](https://github.com/mapbox/node-fontnik/issues/38#issuecomment-47706957).

Don't want to get sidetrack on this right now, but would it be worth experimenting with Glyphy at some point to address these issues with complex fonts?

/cc @kkaefer @ansis @yhahn

### LINKED ISSUE #3691: "Map#is*Loaded" methods
There is a need for a public API that determines whether or not a source is loaded in throttling calls to `GeoJSONSource#setData`. I often find myself using `map.style.sourceCache[sourceId].loaded()` for this purpose. It would be nice to provide a public API instead: `Map#isSourceLoaded(sourceId: string): boolean`

-----

 - add `Map#isSourceLoaded(sourceId)`
 - add `Map#isStyleLoaded()`
 - add `Map#isTilesLoaded()` / `Map#isTileLoaded(coord)` / ...
 - rename `Map#loaded` -> `Map#isLoaded` / `Map#isEverythingLoaded` / ...

### PR DESCRIPTION
**Pull request by mollymerp**
closes #3691 
Still need to add `areTilesLoaded` + ~rename `Map#loaded()` 👉 `Map#isLoaded`~ + testssss

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -769,6 +778,21 @@ class Map extends Camera {
...
 
     /**
+     * Returns a Boolean indicating whether a tile is loaded.
+     *
+     * @param {string} sourceId The ID of the source to be checked.
+     * @param {number} tileCoordId The ID of the source to be checked.
+     * @returns {boolean} A Boolean indicating whether the source is loaded.
+     */
+
+    isTileLoaded(sourceId, tileCoordId){
```
Comment (jfirebaugh) [inline src/ui/map.js]: I think we should hold off making a public `isTileLoaded` method. The number encoding for tile coordinates is a private implementation detail. Also, not all sources have x/y/z tile IDs (GeoJSON, Image, Video, WMS).

**Comment by jfirebaugh** [comment]
> closes #509

Wrong issue number?

File: src/ui/map.js (hunk 1 above)
Comment (mollymerp) [inline src/ui/map.js]: yeah I was wondering about exposing how the TileCoord#id is computed... I'm fine w leaving this out too. I can't think of a user who would care about an individual tile as opposed to all tiles in viewport 💭

**Comment by mollymerp** [comment]
Do you have opinions on renaming `Map#loaded` to `Map#isLoaded` @jfirebaugh ? I feel like the potential burden of the breaking change outweighs the benefit of syntactical consistency 💭

**Comment by jfirebaugh** [comment]
Yeah, tough call. Judging by other methods like `Popup#isOpen` and `*Handler#isEnabled`, it should be `Map#isLoaded`. Maybe we should save it up for a release in which we make other major breaking changes?

**Comment by mollymerp** [comment]
This is ready for review if we're going to hold on the `Map#loaded` ➡️  `Map#isLoaded` change for a more breaky release

**Pull request merged**